packages/synapse-sdk/src/telemetry/singleton.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Telemetry singleton manager
+ *
+ * Provides a single global telemetry instance that can be used by all telemetry
+ * components (fetch wrapper, error tracking, etc.).
```

Comment:
```suggestion
 * Telemetry singleton manager.
 * Sets up and provides a single global telemetry instance that can be used by all telemetry components:
 * - fetch wrapping via fetch-wrapper.ts
 * - error wrapping via errors.ts
 * Synapse SDK consumers (e.g., filecoin-pin) via #getGlobalTelemetry.
 * 
 * #setupGlobalTelemetry is the entry point and #getGlobalTelemetry is the expected access point within Synapse and beyond.
 * 
 * Setting up the fetch wrapper is managed here.
 * Wrapping of error handling is wired in by `src/utils/index.ts` exporting `src/telemetry/errors.ts#createError()`, which wraps `src/utilts/errors.ts`
```

I'm not saying this needs to be the exact wording but I think paints a clearer picture about this class' role.